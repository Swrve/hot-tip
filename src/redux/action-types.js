export const UPDATE_TOOLTIP = 'HOT_TIP::UPDATE_TOOLTIP'
