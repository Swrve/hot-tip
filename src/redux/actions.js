import * as types from './action-types'

export function addTooltip(tip, position) {
  return {
    type: types.UPDATE_TOOLTIP,
    payload: {
      tip,
      visible: true,
      position: Object.assign({}, position)
    }
  }
}

export function clearTooltip() {
  return {
    type: types.UPDATE_TOOLTIP,
    payload: {
      visible: false
    }
  }
}
