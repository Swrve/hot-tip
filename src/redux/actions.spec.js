import {addTooltip, clearTooltip} from './actions'
import * as types from './action-types'

describe('Store', function() {

  describe('HotTip', function() {

    describe('Redux Actions', function() {

      describe('addTooltip', function() {

        it('called with text and position', () => {
          const action = addTooltip('some text', {x: 1, y: 2})

          expect(action).toEqual({
            type: types.UPDATE_TOOLTIP,
            payload: {
              text: 'some text',
              visible: true,
              position: {
                x: 1,
                y: 2
              }
            }
          })
          expect(action).toMatchSnapshot()
        })
      })

      describe('clearTooltip', function() {

        it('when calledcalled', () => {
          const action = clearTooltip()

          expect(action).toEqual({
            type: types.UPDATE_TOOLTIP,
            payload: {
              visible: false
            }
          })
          expect(action).toMatchSnapshot()
        })
      })
    })
  })
})
