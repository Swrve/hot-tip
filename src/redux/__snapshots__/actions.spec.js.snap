// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store HotTip Redux Actions addTooltip called with tip and position 1`] = `
Object {
  "payload": Object {
    "position": Object {
      "x": 1,
      "y": 2,
    },
    "tip": "some tip",
    "visible": true,
  },
  "type": "HOT_TIP::UPDATE_TOOLTIP",
}
`;

exports[`Store HotTip Redux Actions clearTooltip when calledcalled 1`] = `
Object {
  "payload": Object {
    "visible": false,
  },
  "type": "HOT_TIP::UPDATE_TOOLTIP",
}
`;
